xarray/tests/test_backends.py

```diff
@@ -1751,20 +1752,33 @@ class DaskTestAutocloseTrue(DaskTest):
     autoclose = True
 
 
-@network
 @requires_scipy_or_netCDF4
 @requires_pydap
 class PydapTest(TestCase):
+    def convert_to_pydap_dataset(self, original):
+        from pydap.model import GridType, BaseType, DatasetType
+        ds = DatasetType('bears', **original.attrs)
+        for key, var in original.data_vars.items():
+            v = GridType(key)
+            v[key] = BaseType(key, var.values, dimensions=var.dims,
+                              **var.attrs)
+            for d in var.dims:
+                v[d] = BaseType(d, var[d].values)
+            ds[key] = v
+        # check all dims are stored in ds
+        for d in original.coords:
+            ds[d] = BaseType(d, original[d].values, dimensions=(d, ),
+                             **original[d].attrs)
+        return ds
+
     @contextlib.contextmanager
     def create_datasets(self, **kwargs):
-        url = 'http://test.opendap.org/opendap/hyrax/data/nc/bears.nc'
-        actual = open_dataset(url, engine='pydap', **kwargs)
         with open_example_dataset('bears.nc') as expected:
-            # don't check attributes since pydap doesn't serialize them
-            # correctly also skip the "bears" variable since the test DAP
-            # server incorrectly concatenates it.
-            actual = actual.drop('bears')
-            expected = expected.drop('bears')
+            pydap_ds = self.convert_to_pydap_dataset(expected)
+            actual = open_dataset(PydapDataStore(pydap_ds))
+            # TODO solve this workaround:
+            # netcdf converts string to byte not unicode
+            expected['bears'] = expected['bears'].astype(str)
```

Comment:
`netcdf` seems to oepn string-array as byte. Is there any cleaner workaround to solve this?